crates/burn-import/src/onnx/dim_inference.rs

```diff
@@ -251,92 +251,43 @@ fn reduce_mean_update_outputs(node: &mut Node) {
     }
 }
 
-//fn __unsqueeze_shape
-/// Infers the shape of the output from the input and axes
-/// Right now, this should only be called if the rhs is a constant
+/// Update the output tensor dimension based on the "axes" attribute or the second input
 fn unsqueeze_update_output(node: &mut Node) {
-    if node.inputs.len() != 2 {
-        panic!("Unsqueeze: wrong number of inputs");
-    }
-    // get the values while making sure the types are correct
-    let (input, axes) = match (&node.inputs[0].ty, &node.inputs[1].ty) {
-        (ArgType::Tensor(tensor), ArgType::Tensor(_axes)) => (
-            tensor.clone(),
-            match &node.inputs[1].value {
-                Some(value) => match &value {
-                    Data::Int64s(axes) => Some(axes.clone()),
-                    _ => panic!("Unsqueeze: invalid input types"),
-                },
-                None => None,
+    let axes = if node.inputs.len() == 2 {
+        // get the values while making sure the types are correct
+        match &node.inputs[1].value {
+            Some(value) => match value {
+                Data::Int64s(axes) => Some(axes.clone()),
+                _ => panic!("Unsqueeze: invalid input types"),
             },
-        ),
-        _ => panic!("Unsqueeze: invalid input types"),
+            None => None,
+        }
+    } else {
+        node.attrs
+            .iter()
+            .find_map(|(key, value)| match key.as_str() {
+                "axes" => Some(value.clone().into_i64s()),
+                _ => None,
+            })
```

Comment:
For now this is probably fine, but if we want to actually differentiate between opset versions for ops I think we will need to get the opset from the graph metadata.

Reply:
If we need to protect the future changes from accidentally misbehaving, we need lock into specific opset version. Open next version only after certifying with tests of next opset version. This is a full time job which we do not have resource for.

Reply:
Yeah that's also what I was worried about. We don't have enough bandwidth for that 😅 but your PR adds support for an older opset so I was wondering what the motivation for that was since we might not be able to (nor want to) support all possible versions.

Thanks for addressing that offline too. Your approach makes sense.